fast_api_als/utils/adf.py

```diff
@@ -39,10 +40,14 @@ def is_nan(x):
 
 def parse_xml(adf_xml):
     # use exception handling
-    obj = xmltodict.parse(adf_xml)
+    try:
+        obj = xmltodict.parse(adf_xml)
+        logging.info("fn parse_xml : Parsing done")
+    except Exception as e:
+        logging.error(f"Parsing xml failed: {e.message}")
+        raise Exception(f"Parsing xml failed: {e.message}")
```

Comment:
if you're throwing exception here, make sure you're catching this in the code from where you're calling this function.
Check the logic of the function who is calling this, and decide whether you should throw the exception or do something else. in any case handle it accordingly.

Reply:
parse_xml is used in submit_lead. So if there is an exception catch, I am returning an object with status REJECTED along with  the message.